celery/worker/components.py

```diff
@@ -222,7 +222,7 @@ def create(self, w):
             prefetch_count = max(w.max_concurrency, 1) * w.prefetch_multiplier
         else:
             prefetch_count = w.concurrency * w.prefetch_multiplier
-        c = w.consumer = self.instantiate(
+        c = w.consumer = [self.instantiate(
```

Comment:
we also discussed maybe to test with consumermixin having 2 consumers with different URLs and then debugging. its not used in a worker but is a smaller case to find race conditions and fix kombu.hub and other components that may be failing eg amqp transport frame